docs/sql-ref-syntax-qry-select-transform.md

```diff
@@ -0,0 +1,240 @@
+```
+
+### Parameters
+
+* **expression**
+    
+    Specifies a combination of one or more values, operators and SQL functions that results in a value.
+
+* **row_format**    
+
+    Spark uses the `SERDE` clause to specify a custom SerDe for one table. Otherwise, use the `DELIMITED` clause to use the native SerDe and specify the delimiter, escape character, null character and so on.
+
+* **SERDE**
+
+    Specifies a custom SerDe for one table.
+
+* **serde_class**
+
+    Specifies a fully-qualified class name of a custom SerDe.
+
+* **DELIMITED**
+
+    The `DELIMITED` clause can be used to specify the native SerDe and state the delimiter, escape character, null character and so on.
+
+* **FIELDS TERMINATED BY**
+
+    Used to define a column separator.
+    
+* **COLLECTION ITEMS TERMINATED BY**
+
+    Used to define a collection item separator.
+   
+* **MAP KEYS TERMINATED BY**
+
+    Used to define a map key separator.
+    
+* **LINES TERMINATED BY**
+
+    Used to define a row separator.
+    
+* **NULL DEFINED AS**
+
+    Used to define the specific value for NULL.
+    
+* **ESCAPED BY**
+
+    Used for escape mechanism.
+
+* **RECORDWRITER**
+
+    Specifies a fully-qualified class name of a custom RecordWriter. The default value is `org.apache.hadoop.hive.ql.exec.TextRecordWriter`.
+
+* **RECORDREADER**
+
+    Specifies a fully-qualified class name of a custom RecordReader. The default value is `org.apache.hadoop.hive.ql.exec.TextRecordReader`.
+
+* **command_or_script**
+
+    Specifies a command or a path to script to process data.
+
+### SerDe behavior
+
+Spark uses the Hive Serde `org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe` by default, so columns will be casted
+to `STRING` and combined by tabs before feeding to the user script. All `NULL` values will be converted
+to the literal string `"\N"` in order to differentiate `NULL` values from empty strings. The standard output of the
+user script will be treated as tab-separated `STRING` columns, any cell containing only `"\N"` will be re-interpreted
+as a `NULL`, and then the resulting STRING column will be cast to the data type specified in `col_type`. If the actual
+number of output columns is less than the number of specified output columns, insufficient output columns will be
+supplemented with `NULL`. If the actual number of output columns is more than the number of specified output columns,
+the output columns will only select the corresponding columns and the remaining part will be discarded.
+If there is no `AS` clause after `USING my_script`, Spark assumes that the output of the script contains 2 attributes:
+
+   1. key: which is before the first tab.
+   2. value: which is the rest after the first tab.
+
+If there is no enough tab, Spark will return `NULL` value. Note that this is different from specifying an `AS key, value`
+because in that case, the value will only contain the portion between the first tab and the second tab if there are multiple tabs.
```

Comment:
>  Note that this is different from specifying an `AS key, value`
because in that case, the value will only contain the portion between the first tab and the second tab if there are multiple tabs.

I think the statement above looks useless because you've already described the behaviour if the actual number of output columns is more than the number of specified output columns.

Reply:
removed